src/main/lens-proxy/lens-proxy.global-override-for-injectable.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import { getGlobalOverride } from "../../common/test-utils/get-global-override";
+import lensProxyPortInjectable from "./lens-proxy-port.injectable";
+import lensProxyInjectable from "./lens-proxy.injectable";
+
+export default getGlobalOverride(lensProxyInjectable, (di) => {
+  const lensProxyPort = di.inject(lensProxyPortInjectable);
+
+  return ({
+    close: () => { },
+    listen: async () => {
+      lensProxyPort.set(12345);
+    },
```

Comment:
Looks like setup of a temporal dependency in a global override. Is this the case? Some ideas to improve the design:

1. We could instead override the side-effects of `lensProxyInjectable`, and let `lensProxyPort` be set more plausibly.
2. We could consider having `lensProxyPort` add `lensProxy` as dependency. This would mean elimination of the logical/temporal dependency, as then the port would always be true to whatever `lensProxy` says it should be, without state (ie. reactively).
3. Failing at previous stuff (or while waiting), maybe there should also be a `globalOverride` for `lensProxyPort`, instead of this fake implementation.

To know that thing such as `lensProxy` needs to be magically listened before accessing `lensProxyPort` is a big ask.

Reply:
I think instead it shows more confidence in actual business code if we do not do that sort of overriding magic. It means that the `Runnable` phases are setup correctly and verified in unit tests (without the side effects or shared state).